bin/check-s3-object.rb

```diff
@@ -159,17 +159,10 @@ def run
         key_search = config[:key_prefix]
         output = s3.list_objects(bucket: config[:bucket_name], prefix: key_search)
 
-        if output.contents.size.to_i < 1
-          critical "Object with prefix \"#{key_search}\" not found in bucket #{config[:bucket_name]}"
-        end
-
-        if output.contents.size.to_i > 1
-          critical "Your prefix \"#{key_search}\" return too much files, you need to be more specific"
-        end
-
-        key_fullname = output.contents[0].key
-        age = Time.now.to_i - output.contents[0].last_modified.to_i
-        size = output.contents[0].size
+        # Consider most recent object in the list
+        key_fullname = output.contents.last.key
```

Comment:
I am not sure we can rely on the last key to be the oldest. From what I read [here](https://docs.aws.amazon.com/sdkforruby/api/Aws/S3/Client.html#list_objects-instance_method) it will will return up to 1000 objects and a `next_marker` that can be used to make subsequent requests. Also I am not sure that they come back sorted by date and after looking at the example that seems unlikely. We would want to inspect the `last_modified` key in the array of objects to know which one is the newest one.